### ISSUE TITLE
Segfault on Arch Linux

### ISSUE THREAD

**Issue by JuniorJPDJ** [telegramdesktop/tdesktop#16294]
### Steps to reproduce
It happens randomly and often enough to be annoying :<

### Actual behaviour
Telegram crashes with SIGSEGV

### Configuration
**Operating system:**
Arch Linux + KDE
**Version of Telegram Desktop:**
2.7.4
**Installation source (Linux Only)** - the official website / GitHub releases / flatpak / snap / distribution package:
Arch package
**Used theme**:
Night

GDB:
```
Thread 1 "telegram-deskto" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffdd6db180 (LWP 205352)]
0x00007ffff07108a3 in Glib::ustring::compare(Glib::ustring const&) const () from /usr/lib/libglibmm-2.4.so.1
(gdb) c
Continuing.
Couldn't get registers: No such process.
Couldn't get registers: No such process.
(gdb) backtrace 
Python Exception <class 'gdb.error'> Couldn't get registers: No such process.: 
Couldn't get registers: No such process.
Couldn't get registers: No such process.
(gdb) backtrace 
Python Exception <class 'gdb.error'> Couldn't get registers: No such process.: 
Couldn't get registers: No such process.
(gdb) 
```

<details><summary><b>Logs</b>:</summary>
No logs now, I overwrote them by accident. I'll add later.
</details>

**Comment by ilya-fedin** [comment]
Switch to official binary from that buggy package

**Comment by fajfer** [comment]
> Switch to official binary from that buggy package

@Aokromes This is very alarming, do the maintainers know this?

**Comment by ilya-fedin** [comment]
> This is very alarming, do the maintainers know this?

No idea

**Comment by fajfer** [comment]
![ok](https://en.meming.world/images/en/thumb/a/af/Understandable%2C_Have_a_Great_Day.jpg/300px-Understandable%2C_Have_a_Great_Day.jpg)

**Comment by JuniorJPDJ** [comment]
Next trace:
```
Thread 1 "telegram-deskto" received signal SIGSEGV, Segmentation fault.
0x00007fffee37ae00 in ?? () from /usr/lib/libglib-2.0.so.0
(gdb) backtrace
#0  0x00007fffee37ae00 in  () at /usr/lib/libglib-2.0.so.0
#1  0x00007fffee37b347 in g_utf8_collate () at /usr/lib/libglib-2.0.so.0
#2  0x00005555570c7100 in  ()
#3  0x00005555570bf2db in  ()
#4  0x000055555612473f in  ()
#5  0x0000555556124957 in  ()
#6  0x00005555576ca954 in  ()
#7  0x00007fffef1f7762 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#8  0x0000555556790b2f in  ()
#9  0x00007fffefb4781a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#10 0x00007fffefb4a313 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () at /usr/lib/libQt5Core.so.5
#11 0x00007fffefba0ce4 in  () at /usr/lib/libQt5Core.so.5
#12 0x00007fffee34702c in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#13 0x00007fffee39ab59 in  () at /usr/lib/libglib-2.0.so.0
#14 0x00007fffee344781 in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#15 0x00007fffefba0321 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/libQt5Core.so.5
#16 0x00007fffefb4617c in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/libQt5Core.so.5
#17 0x00007fffefb4e634 in QCoreApplication::exec() () at /usr/lib/libQt5Core.so.5
#18 0x000055555678f6c5 in  ()
#19 0x000055555677d3bf in  ()
#20 0x000055555677d685 in  ()
#21 0x0000555555b89a57 in  ()
#22 0x00007fffedcceb25 in __libc_start_main () at /usr/lib/libc.so.6
#23 0x0000555555b9345e in  ()
```

And log is clean.

I'll move to official binary and tell if this still happens.

**Comment by ilya-fedin** [comment]
Well, that crash is not an Arch issue (it's reported every day multiple times, so I don't look at content when I see Arch in issue name), but it's fixed in 2.7.5 beta

**Comment by JuniorJPDJ** [comment]
Yup, just got similar one on official binary:
```
Thread 1 "telegram-deskto" received signal SIGSEGV, Segmentation fault.
0x0000000003335e13 in Glib::ustring::empty() const ()
(gdb) backtrace
#0  0x0000000003335e13 in Glib::ustring::empty() const ()
#1  0x0000000001a6b073 in  ()
#2  0x0000000001a63cab in  ()
#3  0x0000000001398c61 in crl::queue::ProcessCallback(void*) ()
#4  0x00000000011bbd99 in Ui::MainQueueProcessor::event(QEvent*) ()
#5  0x0000000003607e7f in  ()
#6  0x0000000003611598 in  ()
#7  0x000000000218162e in Core::Sandbox::notify(QObject*, QEvent*) ()
#8  0x0000000003f5bf6a in  ()
#9  0x0000000003f5fab5 in  ()
#10 0x0000000003fb5343 in  ()
#11 0x00007ffff790e02c in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#12 0x00007ffff7961b59 in  () at /usr/lib/libglib-2.0.so.0
#13 0x00007ffff790b781 in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#14 0x0000000003fb547e in  ()
#15 0x0000000003f5ae9e in  ()
#16 0x0000000003f63f66 in  ()
#17 0x000000000218758b in Core::Sandbox::start() ()
#18 0x000000000218b198 in Core::Launcher::executeApplication() ()
#19 0x000000000218b335 in Core::Launcher::exec() ()
#20 0x0000000001041508 in main ()
(gdb) c
Continuing.
[New Thread 0x7fff777fe640 (LWP 868576)]
[Thread 0x7fffceffd640 (LWP 865640) exited]
[Thread 0x7fffcffff640 (LWP 865639) exited]
[Thread 0x7fff39f32640 (LWP 865277) exited]
[Thread 0x7fffce7fc640 (LWP 865275) exited]
[Thread 0x7ffecf7fe640 (LWP 865276) exited]
[Thread 0x7fffcf7fe640 (LWP 865274) exited]
[Thread 0x7fff77fff640 (LWP 868495) exited]
[Thread 0x7fffcdffb640 (LWP 865641) exited]
[Thread 0x7fffed6a4640 (LWP 865278) exited]
[New Thread 0x7fff77fff640 (LWP 868578)]
[Thread 0x7fff77fff640 (LWP 868578) exited]
[Thread 0x7fff777fe640 (LWP 868576) exited]

Thread 1 "telegram-deskto" received signal SIGSEGV, Segmentation fault.
0x0000000003335e13 in Glib::ustring::empty() const ()
(gdb) c
Continuing.
Couldn't get registers: No such process.
Couldn't get registers: No such process.
```

So you say It's fixed in beta? I'm waiting for full release then :D
